Cannot find tinyxml during windows build

At conda-forge, I'm trying to build this package with windows (win64) support.  We are currently using a quasi-similar build script for all platforms (sample here: https://github.com/anjos/libmediainfo-feedstock/tree/autotick-update/recipe).  This works well for libzen (https://github.com/conda-forge/libzen-feedstock).

However, for this library, cmake cannot find an installed version of tinyxml2 (log file: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=406414&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=f5d15007-a01c-5ad8-c9ce-4d519d3b275f&l=429).  I double-checked, both `pkg-config` and `tinyxml2` are installed on the build environment.  All other packages are correctly detected (ZenLib, ZLIB, CURL).

I wonder if the conda-forge packaged version of tinyxml2 (https://anaconda.org/conda-forge/tinyxml2/9.0.0/download/win-64/tinyxml2-9.0.0-h39d44d4_1.tar.bz2) is somehow incompatible with `Project/CMake/cmake/modules/FindTinyXML.cmake` and why would that be the case.  tinyxml2 looks OK from first sight and includes a pkg-config file as well.

Are there any suggestions I could explore to find a fix for this?

Here is my summary of what is going on, after several attempts and a success.  The cmake build is setup to search for a system version of tinyxml2 and fail quietly, resorting to a locally shipped version of this package.  This does not work under conda-forge/win-64 because:

* The fallback version of tinyxml2 shipped within this package does not compile well under windows, it misses some symbols, and fails during linking at the conda-forge setup.  The missing symbols are:
  * `error LNK2001: unresolved external symbol "protected: virtual void __cdecl tinyxml2::XMLPrinter::Print(char const *,...)" (?Print@XMLPrinter@tinyxml2@@MEAAXPEBDZZ)`
  * `error LNK2001: unresolved external symbol "protected: virtual void __cdecl tinyxml2::XMLPrinter::Write(char const *,unsigned __int64)" (?Write@XMLPrinter@tinyxml2@@MEAAXPEBD_K@Z)`
  * `error LNK2001: unresolved external symbol "protected: virtual void __cdecl tinyxml2::XMLPrinter::Putc(char)" (?Putc@XMLPrinter@tinyxml2@@MEAAXD@Z)`
* There is a cmake extension module called `FindTinyXML` which is shipped with this package, but this module explicitly disables the search for tinyxml2 under windows, which implies that under that OS setup we shall always resort to the fallback version of this library which is bogus
* Even if you remove the WIN32 restriction in that extension, so that it starts working under Windows, for conda-forge, the location of `tinyxml2.pc` is atypical.  This location is set correctly by the build infrastructure.  However, MediaInfoLib requires a minimum version of cmake to be smaller than 3.1 (2.8.11, to be precise).  This has a funny implication on the setup of the search path for pkg-config (which is kind of explained here: https://cmake.org/cmake/help/latest/module/FindPkgConfig.html).  If the minimum cmake version is set less than 3.1, then cmake by default removes access to any path preset by the build infrastructure and rely only on what pkg-config itself can find on their default search path.  Therefore, it will continue to fail under those conditions.

Given this situation, I'd recommend the following changes (I can prepare a PR with the ones you find relevant):

1. Bump the minimum required version of cmake from 2.8.11 to 3.1.  (this may have implications to libzen and mediainfo).   Instead, you may set `PKG_CONFIG_USE_CMAKE_PREFIX_PATH=ON` which will mitigate this problem, avoiding Cmake sets it to OFF because the minimum required version is smaller than 3.1.
2. Modify `FindTinyXML.cmake` to remove the `WIN32` flag restriction
3. Modify `FindTinyXML.cmake` to remove the `QUIET` flag, and let the process print the found version of tinyxml2 if available on the system and be more explicit of what is going on.
4. I'd also recommend updating the shipped, fallback version of tinyxml2 to avoid compilation issues under windows.  However, this is is untested.

Let me know which path you prefer to follow.

--- a/Project/CMake/cmake/modules/FindTinyXML.cmake
+++ b/Project/CMake/cmake/modules/FindTinyXML.cmake
@@ -8,20 +8,18 @@
 #
 # CopyPolicy: Released under the terms of the LGPLv2.1 or later
 
-if(NOT WIN32)
-  find_package(PkgConfig)
-  if(PKG_CONFIG_FOUND)
-    if(TinyXML_FIND_VERSION)
-      if(TinyXML_FIND_VERSION_EXACT)
-        pkg_check_modules(PC_TINYXML QUIET tinyxml2=${TinyXML_FIND_VERSION})
-      else(TinyXML_FIND_VERSION_EXACT)
-        pkg_check_modules(PC_TINYXML QUIET tinyxml2>=${TinyXML_FIND_VERSION})
-      endif(TinyXML_FIND_VERSION_EXACT)
-    else(TinyXML_FIND_VERSION)
-      pkg_check_modules(PC_TINYXML QUIET tinyxml2)
-    endif(TinyXML_FIND_VERSION)
-  endif(PKG_CONFIG_FOUND)
-endif(NOT WIN32)
+find_package(PkgConfig)
+if(PKG_CONFIG_FOUND)
+  if(TinyXML_FIND_VERSION)
+    if(TinyXML_FIND_VERSION_EXACT)
+      pkg_check_modules(PC_TINYXML tinyxml2=${TinyXML_FIND_VERSION})
+    else(TinyXML_FIND_VERSION_EXACT)
+      pkg_check_modules(PC_TINYXML tinyxml2>=${TinyXML_FIND_VERSION})
+    endif(TinyXML_FIND_VERSION_EXACT)
+  else(TinyXML_FIND_VERSION)
+    pkg_check_modules(PC_TINYXML tinyxml2)
+  endif(TinyXML_FIND_VERSION)
+endif(PKG_CONFIG_FOUND)
 
 set(TinyXML_INCLUDE_DIRS ${PC_TINYXML_INCLUDE_DIRS} CACHE PATH "TinyXML include directory" FORCE)
 set(TinyXML_LIBRARY_DIRS ${PC_TINYXML_LIBRARY_DIRS} CACHE PATH "TinyXML library directory" FORCE)
@@ -34,4 +32,4 @@ find_package_handle_standard_args(TinyXML
   )
 
 set(TinyXML_FOUND ${TINYXML_FOUND})
-mark_as_advanced(TinyXML_INCLUDE_DIRS TinyXML_LIBRARY_DIRS TinyXML_LIBRARIES) 
+mark_as_advanced(TinyXML_INCLUDE_DIRS TinyXML_LIBRARY_DIRS TinyXML_LIBRARIES)

